src/mailadm/conn.py

```diff
@@ -232,6 +232,46 @@ def get_expired_users(self, sysdate):
             users.append(UserInfo(*args))
         return users
 
+    def get_users_to_warn(self, sysdate: int) -> [{}]:
+        allusers = self.get_user_list()
+        users_to_warn = []
+        for user in allusers:
+            if user.token_name == "created in mailcow":
+                continue
+            year = 31536000
+            month = 2592000
+            week = 604800
+            day = 86400
+            warnmsg = "Your account will expire in {}."
+            timeleft = ""
+            if user.ttl >= year:
+                if user.warned == 0 and user.date + user.ttl < sysdate + month:
+                    timeleft = "30 days"
+                elif user.warned == 1 and user.date + user.ttl < sysdate + week:
+                    timeleft = "7 days"
+                elif user.warned == 2 and user.date + user.ttl < sysdate + day:
+                    timeleft = "1 day"
```

Comment:
only occurs to me now (sorry) but what if for some reason the bot doesn't run for a few days, and then performs this expiry check?  Will we not send a message "expired in 7 days" but it might only be "4 days" actually? 
We are only ever warning about "X days" so maybe there can be helper function that computers the number of remaining days irrespective how often expiry-checking is performed?

Reply:
Maybe the message should be "You account will expire at UTC-DATE, about N days from now."

Reply:
good point. Problem: imo we can't expect users to know in which timezone they are. Maybe we can avoid timezone specifics? for most users a date would be enough, and if the time is shorter, just saying how many hours it will take should be enough.

Reply:
sure, we are anyway missing localization as well.